File: app/src/main/java/lajevski/radoslav/tesonetparty/di/ApplicationComponent.java
package lajevski.radoslav.tesonetparty.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import lajevski.radoslav.tesonetparty.App;
import lajevski.radoslav.tesonetparty.data.DataManager;
import lajevski.radoslav.tesonetparty.data.network.NetworkModule;

/**
 * Created by Radoslav on 1/16/2018.
 */

@Singleton
@Component(modules = {ApplicationModule.class, NetworkModule.class})
public interface ApplicationComponent {

    void inject(App app);

    @ApplicationContext
    Context context();

    Application application();

    DataManager getDataManager();
}

File: app/src/main/java/com/example/ievazygaite/androidparty/di/component/ApplicationComponent.kt
package com.example.ievazygaite.androidparty.di.component

import com.example.ievazygaite.androidparty.BaseApplication
import com.example.ievazygaite.androidparty.di.module.ApplicationModule
import dagger.Component

@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {
    fun inject(application: BaseApplication)
}